packages-content-model/roosterjs-content-model-plugins/lib/edit/keyboardTab.ts

```diff
@@ -0,0 +1,61 @@
+import { getOperationalBlocks, isBlockGroupOfType } from 'roosterjs-content-model-core';
+import { setIndentation } from 'roosterjs-content-model-api';
+
+import type {
+    ContentModelDocument,
+    ContentModelListItem,
+    DOMSelection,
+    IStandaloneEditor,
+} from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function keyboardTab(editor: IStandaloneEditor, rawEvent: KeyboardEvent) {
+    const selection = editor.getDOMSelection();
+
+    if (shouldHandleTab(rawEvent, selection)) {
+        editor.takeSnapshot();
+
+        editor.formatContentModel(
+            (model, _context) => {
+                return handleTabOnList(editor, model, rawEvent);
+            },
+            {
+                rawEvent,
+            }
+        );
+
+        return true;
+    }
+}
+
+function shouldHandleTab(rawEvent: KeyboardEvent, selection: DOMSelection | null) {
+    return rawEvent.key == 'Tab' && selection && selection?.type == 'range';
```

Comment:
we have already checked key == 'Tab' before entering this function, so no need to check it again here

Reply:
no need to check "selection" since you already use "selection?" to check type